[Swift in WebKit] Work towards fixing the WTF clang module

rr-codes opened the pull request:
#### a37b18497f9a36e072449253c8580acfcb87bd34
<pre>
[Swift in WebKit] Work towards fixing the WTF clang module
<a href="https://bugs.webkit.org/show_bug.cgi?id=304457">https://bugs.webkit.org/show_bug.cgi?id=304457</a>
<a href="https://rdar.apple.com/166842178">rdar://166842178</a>

Reviewed by Megan Gardner.

Add some missing header includes across WTF.

* Source/WTF/wtf/AnsiColors.h:
* Source/WTF/wtf/AutodrainedPool.h:
* Source/WTF/wtf/BubbleSort.h:
* Source/WTF/wtf/ByteOrder.h:
* Source/WTF/wtf/ClockType.h:
* Source/WTF/wtf/CryptographicUtilities.h:
* Source/WTF/wtf/CryptographicallyRandomNumber.h:
* Source/WTF/wtf/Dominators.h:
* Source/WTF/wtf/DoublyLinkedList.h:
* Source/WTF/wtf/EnumTraits.h:
* Source/WTF/wtf/EnumeratedArray.h:
* Source/WTF/wtf/Expected.h:
* Source/WTF/wtf/ExperimentalFeatureNames.h:
* Source/WTF/wtf/FastTLS.h:
* Source/WTF/wtf/FlipBytes.h:
* Source/WTF/wtf/FunctionTraits.h:
* Source/WTF/wtf/IndexKeyType.h:
* Source/WTF/wtf/IndexedContainerIterator.h:
* Source/WTF/wtf/IndexedRange.h:
* Source/WTF/wtf/Insertion.h:
* Source/WTF/wtf/InstanceCounted.h:
* Source/WTF/wtf/Int128.h:
* Source/WTF/wtf/IteratorRange.h:
* Source/WTF/wtf/LEBDecoder.h:
* Source/WTF/wtf/LLVMProfilingUtils.h:
* Source/WTF/wtf/LoggerHelper.h:
* Source/WTF/wtf/LoggingHashMap.h:
* Source/WTF/wtf/LoggingHashSet.h:
* Source/WTF/wtf/MachSendRight.h:
* Source/WTF/wtf/MappedFileData.h:
* Source/WTF/wtf/MemoryDump.h:
* Source/WTF/wtf/MemoryFootprint.h:
* Source/WTF/wtf/MmapSpan.h:
* Source/WTF/wtf/NumberOfCores.h:
* Source/WTF/wtf/OSAllocator.h:
* Source/WTF/wtf/OSRandomSource.h:
* Source/WTF/wtf/OverflowPolicy.h:
* Source/WTF/wtf/Packed.h:
* Source/WTF/wtf/ParallelJobsGeneric.h:
* Source/WTF/wtf/ParallelJobsLibdispatch.h:
* Source/WTF/wtf/ParallelJobsOpenMP.h:
* Source/WTF/wtf/Platform.h:
* Source/WTF/wtf/PlatformEnable.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WTF/wtf/PointerPreparations.h:
* Source/WTF/wtf/ProcessID.h:
* Source/WTF/wtf/RAMSize.h:
* Source/WTF/wtf/RandomDevice.h:
* Source/WTF/wtf/RawHex.h:
* Source/WTF/wtf/RawPointer.h:
* Source/WTF/wtf/RawValueTraits.h:
* Source/WTF/wtf/RecursableLambda.h:
* Source/WTF/wtf/ResourceUsage.h:
* Source/WTF/wtf/RuntimeApplicationChecks.h:
* Source/WTF/wtf/SIMDHelpers.h:
* Source/WTF/wtf/SafeStrerror.h:
* Source/WTF/wtf/SaturatedArithmetic.h:
* Source/WTF/wtf/ScopedPrintStream.h:
* Source/WTF/wtf/SetForScope.h:
* Source/WTF/wtf/SimpleStats.h:
* Source/WTF/wtf/SixCharacterHash.h:
* Source/WTF/wtf/SoftLinking.h:
* Source/WTF/wtf/StackPointer.h:
* Source/WTF/wtf/StackStats.h:
* Source/WTF/wtf/StackTrace.h:
* Source/WTF/wtf/StdFilesystem.h:
* Source/WTF/wtf/StdIntExtras.h:
* Source/WTF/wtf/Stopwatch.h:
* Source/WTF/wtf/SystemMalloc.h:
* Source/WTF/wtf/SystemTracing.h:
* Source/WTF/wtf/TaggedPtr.h:
* Source/WTF/wtf/ThreadAssertions.h:
* Source/WTF/wtf/TranslatedProcess.h:
* Source/WTF/wtf/TriState.h:
* Source/WTF/wtf/URLHelpers.h:
* Source/WTF/wtf/Unexpected.h:
* Source/WTF/wtf/UniqueRef.h:
* Source/WTF/wtf/VMTags.h:
* Source/WTF/wtf/WTFProcess.h:
* Source/WTF/wtf/WeakRandom.h:
* Source/WTF/wtf/WeakRandomNumber.h:
* Source/WTF/wtf/WindowsExtras.h:
* Source/WTF/wtf/ZippedRange.h:
* Source/WTF/wtf/cf/CFURLExtras.h:
* Source/WTF/wtf/cf/NotificationCenterCF.h:
* Source/WTF/wtf/cf/TypeCastsCF.h:
* Source/WTF/wtf/cf/VectorCF.h:
* Source/WTF/wtf/cocoa/AuditToken.h:
* Source/WTF/wtf/cocoa/CrashReporter.h:
* Source/WTF/wtf/cocoa/Entitlements.h:
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.h:
* Source/WTF/wtf/cocoa/SoftLinking.h:
* Source/WTF/wtf/darwin/OSLogPrintStream.h:
* Source/WTF/wtf/dragonbox/detail/decimal_fp.h:
* Source/WTF/wtf/ios/WebCoreThread.h:
* Source/WTF/wtf/spi/cf/CFBundleSPI.h:
* Source/WTF/wtf/spi/cf/CFPrivSPI.h:
* Source/WTF/wtf/spi/cf/CFRunLoopSPI.h:
* Source/WTF/wtf/spi/cf/CFStringSPI.h:
* Source/WTF/wtf/spi/cocoa/BOMSPI.h:
* Source/WTF/wtf/spi/cocoa/CFXPCBridgeSPI.h:
* Source/WTF/wtf/spi/cocoa/CrashReporterClientSPI.h:
* Source/WTF/wtf/spi/cocoa/IOReturnSPI.h:
* Source/WTF/wtf/spi/cocoa/IOSurfaceSPI.h:
* Source/WTF/wtf/spi/cocoa/IOTypesSPI.h:
* Source/WTF/wtf/spi/cocoa/MachVMSPI.h:
* Source/WTF/wtf/spi/cocoa/NSLocaleSPI.h:
* Source/WTF/wtf/spi/cocoa/NSObjCRuntimeSPI.h:
* Source/WTF/wtf/spi/cocoa/OSLogSPI.h:
* Source/WTF/wtf/spi/cocoa/SecuritySPI.h:
* Source/WTF/wtf/spi/cocoa/XTSPI.h:
* Source/WTF/wtf/spi/cocoa/objcSPI.h:
* Source/WTF/wtf/spi/darwin/CodeSignSPI.h:
* Source/WTF/wtf/spi/darwin/DataVaultSPI.h:
* Source/WTF/wtf/spi/darwin/DispatchSPI.h:
* Source/WTF/wtf/spi/darwin/OSVariantSPI.h:
* Source/WTF/wtf/spi/darwin/ProcessMemoryFootprint.h:
* Source/WTF/wtf/spi/darwin/SandboxSPI.h:
* Source/WTF/wtf/spi/darwin/XPCSPI.h:
* Source/WTF/wtf/spi/darwin/dyldSPI.h:
* Source/WTF/wtf/spi/mac/MetadataSPI.h:
* Source/WTF/wtf/text/FastCharacterComparison.h:
* Source/WTF/wtf/text/LineEnding.h:
* Source/WTF/wtf/text/TextBreakIteratorInternalICU.h:
* Source/WTF/wtf/text/UTF8ConversionError.h:
* Source/WTF/wtf/text/cf/TextBreakIteratorCFStringTokenizer.h:
* Source/WTF/wtf/text/icu/UTextProvider.h:
* Source/WTF/wtf/text/icu/UTextProviderLatin1.h:
* Source/WTF/wtf/text/icu/UTextProviderUTF16.h:
* Source/WTF/wtf/text/icu/UnicodeExtras.h:
* Source/WTF/wtf/unicode/Collator.h:
* Source/WTF/wtf/unicode/UTF8Conversion.h:
* Source/WTF/wtf/unicode/icu/ICUHelpers.h:

Canonical link: <a href="https://commits.webkit.org/304750@main">https://commits.webkit.org/304750@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3b504289840888ddf23256008a27933da3d6cdf3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/136415 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/8772 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/47695 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/144127 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/89386 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/151be280-937f-49a0-a679-b84571cf6ac0) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/9455 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/8616 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/104317 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/89386 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/903625e8-df45-448b-9295-0f89b9368dc1) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/139360 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/6896 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/122231 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/85153 "Passed tests") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/eb532d27-291b-401a-9d05-e10e3db5e077) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/6540 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/4199 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/4719 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/128372 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/115838 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/40437 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/146874 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/134899 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/8454 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/41006 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/112654 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/8471 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/7104 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/113000 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28694 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/6477 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/118541 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/62441 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/8502 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/36593 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/167678 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/8220 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/72061 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/43739 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/8442 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/8294 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

megangardner reviewed (approved):
Please fix style errors. Otherwise r+

rr-codes committed:
[Swift in WebKit] Work towards fixing the WTF clang module
https://bugs.webkit.org/show_bug.cgi?id=304457
rdar://166842178

Reviewed by Megan Gardner.

Add some missing header includes across WTF.

* Source/WTF/wtf/AnsiColors.h:
* Source/WTF/wtf/AutodrainedPool.h:
* Source/WTF/wtf/BubbleSort.h:
* Source/WTF/wtf/ByteOrder.h:
* Source/WTF/wtf/ClockType.h:
* Source/WTF/wtf/CryptographicUtilities.h:
* Source/WTF/wtf/CryptographicallyRandomNumber.h:
* Source/WTF/wtf/Dominators.h:
* Source/WTF/wtf/DoublyLinkedList.h:
* Source/WTF/wtf/EnumTraits.h:
* Source/WTF/wtf/EnumeratedArray.h:
* Source/WTF/wtf/Expected.h:
* Source/WTF/wtf/ExperimentalFeatureNames.h:
* Source/WTF/wtf/FastTLS.h:
* Source/WTF/wtf/FlipBytes.h:
* Source/WTF/wtf/FunctionTraits.h:
* Source/WTF/wtf/IndexKeyType.h:
* Source/WTF/wtf/IndexedContainerIterator.h:
* Source/WTF/wtf/IndexedRange.h:
* Source/WTF/wtf/Insertion.h:
* Source/WTF/wtf/InstanceCounted.h:
* Source/WTF/wtf/Int128.h:
* Source/WTF/wtf/IteratorRange.h:
* Source/WTF/wtf/LEBDecoder.h:
* Source/WTF/wtf/LLVMProfilingUtils.h:
* Source/WTF/wtf/LoggerHelper.h:
* Source/WTF/wtf/LoggingHashMap.h:
* Source/WTF/wtf/LoggingHashSet.h:
* Source/WTF/wtf/MachSendRight.h:
* Source/WTF/wtf/MappedFileData.h:
* Source/WTF/wtf/MemoryDump.h:
* Source/WTF/wtf/MemoryFootprint.h:
* Source/WTF/wtf/MmapSpan.h:
* Source/WTF/wtf/NumberOfCores.h:
* Source/WTF/wtf/OSAllocator.h:
* Source/WTF/wtf/OSRandomSource.h:
* Source/WTF/wtf/OverflowPolicy.h:
* Source/WTF/wtf/Packed.h:
* Source/WTF/wtf/ParallelJobsGeneric.h:
* Source/WTF/wtf/ParallelJobsLibdispatch.h:
* Source/WTF/wtf/ParallelJobsOpenMP.h:
* Source/WTF/wtf/Platform.h:
* Source/WTF/wtf/PlatformEnable.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WTF/wtf/PointerPreparations.h:
* Source/WTF/wtf/ProcessID.h:
* Source/WTF/wtf/RAMSize.h:
* Source/WTF/wtf/RandomDevice.h:
* Source/WTF/wtf/RawHex.h:
* Source/WTF/wtf/RawPointer.h:
* Source/WTF/wtf/RawValueTraits.h:
* Source/WTF/wtf/RecursableLambda.h:
* Source/WTF/wtf/ResourceUsage.h:
* Source/WTF/wtf/RuntimeApplicationChecks.h:
* Source/WTF/wtf/SIMDHelpers.h:
* Source/WTF/wtf/SafeStrerror.h:
* Source/WTF/wtf/SaturatedArithmetic.h:
* Source/WTF/wtf/ScopedPrintStream.h:
* Source/WTF/wtf/SetForScope.h:
* Source/WTF/wtf/SimpleStats.h:
* Source/WTF/wtf/SixCharacterHash.h:
* Source/WTF/wtf/SoftLinking.h:
* Source/WTF/wtf/StackPointer.h:
* Source/WTF/wtf/StackStats.h:
* Source/WTF/wtf/StackTrace.h:
* Source/WTF/wtf/StdFilesystem.h:
* Source/WTF/wtf/StdIntExtras.h:
* Source/WTF/wtf/Stopwatch.h:
* Source/WTF/wtf/SystemMalloc.h:
* Source/WTF/wtf/SystemTracing.h:
* Source/WTF/wtf/TaggedPtr.h:
* Source/WTF/wtf/ThreadAssertions.h:
* Source/WTF/wtf/TranslatedProcess.h:
* Source/WTF/wtf/TriState.h:
* Source/WTF/wtf/URLHelpers.h:
* Source/WTF/wtf/Unexpected.h:
* Source/WTF/wtf/UniqueRef.h:
* Source/WTF/wtf/VMTags.h:
* Source/WTF/wtf/WTFProcess.h:
* Source/WTF/wtf/WeakRandom.h:
* Source/WTF/wtf/WeakRandomNumber.h:
* Source/WTF/wtf/WindowsExtras.h:
* Source/WTF/wtf/ZippedRange.h:
* Source/WTF/wtf/cf/CFURLExtras.h:
* Source/WTF/wtf/cf/NotificationCenterCF.h:
* Source/WTF/wtf/cf/TypeCastsCF.h:
* Source/WTF/wtf/cf/VectorCF.h:
* Source/WTF/wtf/cocoa/AuditToken.h:
* Source/WTF/wtf/cocoa/CrashReporter.h:
* Source/WTF/wtf/cocoa/Entitlements.h:
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.h:
* Source/WTF/wtf/cocoa/SoftLinking.h:
* Source/WTF/wtf/darwin/OSLogPrintStream.h:
* Source/WTF/wtf/dragonbox/detail/decimal_fp.h:
* Source/WTF/wtf/ios/WebCoreThread.h:
* Source/WTF/wtf/spi/cf/CFBundleSPI.h:
* Source/WTF/wtf/spi/cf/CFPrivSPI.h:
* Source/WTF/wtf/spi/cf/CFRunLoopSPI.h:
* Source/WTF/wtf/spi/cf/CFStringSPI.h:
* Source/WTF/wtf/spi/cocoa/BOMSPI.h:
* Source/WTF/wtf/spi/cocoa/CFXPCBridgeSPI.h:
* Source/WTF/wtf/spi/cocoa/CrashReporterClientSPI.h:
* Source/WTF/wtf/spi/cocoa/IOReturnSPI.h:
* Source/WTF/wtf/spi/cocoa/IOSurfaceSPI.h:
* Source/WTF/wtf/spi/cocoa/IOTypesSPI.h:
* Source/WTF/wtf/spi/cocoa/MachVMSPI.h:
* Source/WTF/wtf/spi/cocoa/NSLocaleSPI.h:
* Source/WTF/wtf/spi/cocoa/NSObjCRuntimeSPI.h:
* Source/WTF/wtf/spi/cocoa/OSLogSPI.h:
* Source/WTF/wtf/spi/cocoa/SecuritySPI.h:
* Source/WTF/wtf/spi/cocoa/XTSPI.h:
* Source/WTF/wtf/spi/cocoa/objcSPI.h:
* Source/WTF/wtf/spi/darwin/CodeSignSPI.h:
* Source/WTF/wtf/spi/darwin/DataVaultSPI.h:
* Source/WTF/wtf/spi/darwin/DispatchSPI.h:
* Source/WTF/wtf/spi/darwin/OSVariantSPI.h:
* Source/WTF/wtf/spi/darwin/ProcessMemoryFootprint.h:
* Source/WTF/wtf/spi/darwin/SandboxSPI.h:
* Source/WTF/wtf/spi/darwin/XPCSPI.h:
* Source/WTF/wtf/spi/darwin/dyldSPI.h:
* Source/WTF/wtf/spi/mac/MetadataSPI.h:
* Source/WTF/wtf/text/FastCharacterComparison.h:
* Source/WTF/wtf/text/LineEnding.h:
* Source/WTF/wtf/text/TextBreakIteratorInternalICU.h:
* Source/WTF/wtf/text/UTF8ConversionError.h:
* Source/WTF/wtf/text/cf/TextBreakIteratorCFStringTokenizer.h:
* Source/WTF/wtf/text/icu/UTextProvider.h:
* Source/WTF/wtf/text/icu/UTextProviderLatin1.h:
* Source/WTF/wtf/text/icu/UTextProviderUTF16.h:
* Source/WTF/wtf/text/icu/UnicodeExtras.h:
* Source/WTF/wtf/unicode/Collator.h:
* Source/WTF/wtf/unicode/UTF8Conversion.h:
* Source/WTF/wtf/unicode/icu/ICUHelpers.h:

Canonical link: https://commits.webkit.org/304750@main
